File: server/src/main/java/com/example/server/server/model/Response.java
package com.example.server.server.model;

import java.time.LocalDateTime;
import java.util.Map;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

import org.springframework.http.HttpStatus;


import lombok.Data;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder
@JsonInclude(value = Include.NON_NULL)
public class Response {
    protected LocalDateTime timeStamp;
    protected int statusCode;
    protected HttpStatus status;
    protected String reason;
    protected String message;
    protected String developerMessage;
    protected Map<?, ?> data;
}

File: frontserver/FrontServer/src/app/interface/response.ts
import { Status } from "../enum/status.enum";
import { Server } from "./server";

export interface Response{
     timeStamp : Date;
      statusCode  : number;
     status : string ;
      reason:string;
      message:string;
      developerMessage:string;
    data:{servers?:Server[],server?: Server};
}